libqtile/layout/floating.py

```diff
@@ -88,19 +86,41 @@ def __init__(self, float_rules=None, no_reposition_match=None, **config):
         the screen.
         """
         Layout.__init__(self, **config)
+        self.add_defaults(Floating.defaults)
         self.clients = []
         self.focused = None
         self.group = None
-        self.float_rules = float_rules or DEFAULT_FLOAT_RULES
         self.no_reposition_match = no_reposition_match
-        self.add_defaults(Floating.defaults)
+
+        if override_default_float_rules:
+            self.float_rules = defaultdict(set)
+        else:
+            self.float_rules = DEFAULT_FLOAT_RULES
+        # provide support for deprecated `auto_float_types` argument
+        if self.auto_float_types:
+            logger.warning("`auto_float_types` is deprecated. Use `float_rules` for this.")
+            self.float_rules["wmtype"].update(self.auto_float_types)
+        # provide support for the deprecated list-structure of `float_rules`
+        if isinstance(float_rules, list):
+            logger.warning("The list structure of `float_rules` is deprecated. Use a dictionary instead")
```

Comment:
Let's add a description of how to update your config to /CHANGELOG

Reply:
Will do!